workflow/rules/intersect/extract-wind.smk

```diff
@@ -0,0 +1,194 @@
+    Output:
+        remaining elements: list [list of remaining regions, list of remaining samples, list of remaining nh]
+    """
+    region_all = rsn[0]
+    sample_all = rsn[1]
+    nh_all = rsn[2]
+    nh_completed_files = []
+    for region in region_all:
+        nh_completed_files += glob(
+            os.path.join(
+                config['data_dir'], "intersection", "storm_data", "all_winds", region, "*csv"
+            )
+        )  # find which wind speeds have been completed already
+    nh_completed = [
+        file[file.find("_n") + 2 : -4] for file in nh_completed_files
+    ]  # single out nh identifiers from directory files
+    indices_remaining = [
+        x for x, nh in enumerate(nh_all) if nh not in nh_completed
+    ]  # list all indices that are not already completed
+    region_remaining = [
+        r for x, r in enumerate(region_all) if x in indices_remaining
+    ]  # select based on indices
+    sample_remaining = [
+        s for x, s in enumerate(sample_all) if x in indices_remaining
+    ]  # select based on indices
+    nh_remaining = [
+        nh for x, nh in enumerate(nh_all) if x in indices_remaining
+    ]  # select based on indices
+    print("Remaining storms to evaluate wind speeds:", nh_remaining)
+    return [region_remaining, sample_remaining, nh_remaining]
+
+
+def find_nh(years_inp, region, sample):
+    """Returns a list of all hurricane unique identifiers (nh) in the years years_inp for the given region and sample
+    Input:
+        years_inp: list of years to check within
+        region: str
+        sample: str
+    Output:
+        list of unique nh
+    """
+    filename = os.path.join(
+        "data",
+        "stormtracks",
+        "events",
+        f"STORM_DATA_IBTRACS_{region}_1000_YEARS_{sample}.txt",
+    )
+    if (
+        type(years_inp) != list
+    ):  # if years_inp is not a list, make it a list of length one
+        years_inp = [years_inp]
+    try:
+        df = pd.read_csv(filename, header=None)
+        df.columns = [
+            "year",
+            "month",
+            "number",
+            "step",
+            "basin",
+            "lat",
+            "lon",
+            "pressure",
+            "wind",
+            "radius",
+            "cat",
+            "landfall",
+            "dis_land",
+        ]
+        years_inp = [int(year) for year in years_inp]  # ensure integer type
+        df = df[
+            df["year"].astype(int).isin(years_inp)
+        ]  # filter for selected years (years_inp)
+        df["nh_nosample"] = (
+            df["year"].astype(int).astype(str)
+            + "_"
+            + df["number"].astype(int).astype(str)
+        )  # new column #_# (first # is year, second # is the storm number of that year)
+        nums = list(df["nh_nosample"].unique())  # find all unique #_#
+        ret_lst = [
+            f"{sample}_{num}" for num in nums
+        ]  # add sample for #_#_# (sample is now first #)
```

Comment:
Yes this is an issue i also came across and use same solution idea. I try to use the format r{region}_s{sample}_n{nh} etc to avoid that issue. But nh is the storm number which is a wildcard (rather than consisting of multiple wildcards), so it shouldn't be a problem. I may look into adding this for clarity regardless.